src/main/java/org/dataloader/DataLoaderRegistry.java

```diff
@@ -142,8 +149,17 @@ public void dispatchAll() {
      */
     public int dispatchAllWithCount() {
         int sum = 0;
+        List<CompletableFuture<?>> futuresToDispatch = new ArrayList<>();
         for (DataLoader<?, ?> dataLoader : getDataLoaders()) {
-            sum += dataLoader.dispatchWithCounts().getKeysCount();
+            if (dataLoader.dispatchDepth() > 0) {
+                DispatchResult<?> dispatchResult = dataLoader.dispatchWithCounts();
+                sum += dispatchResult.getKeysCount();
+                futuresToDispatch.add(dispatchResult.getPromisedResults());
+            }
+        }
+        if (futuresToDispatch.size() > 0) {
+            CompletableFuture.allOf(futuresToDispatch.toArray(new CompletableFuture[0])).join();
```

Comment:
`join()` is typically in an async system.  You going to make them sync onto each other